doc/mdc-migration.md

```diff
@@ -0,0 +1,54 @@
+# MDC Android migration
+## Status:
+Accepted on 2020-04-16
+
+## Context:
+We have been using app compat design theme from the start, while that worked perfectly it did not allow us to use new material components because those are dependent on `Theme.MaterialComponents.*` app theme. Using new material components allows us to have much more control over component theming like text styles, icon & icon gravity, strokes, corners, etc.,
```

Comment:
Can we replace the word "app compat" with `Appcompat` since it's a defined package in Android.

Reply:
I have replaced it with `Theme.Design.*` because there are 2 variants of themes in the appcompat package. So I made specific to what we used before.